clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/PushProviders.java

```diff
@@ -518,70 +478,84 @@ public Void call() {
+            jobScheduler.cancel(existingJobId);
+            StorageHelper.remove(context, Constants.PF_JOB_ID);
+        }
+    }
 
+    private void createOrResetWorker(boolean isPingFrequencyUpdated) {
         //Disable push amp for devices below Api 26
         if (VERSION.SDK_INT < VERSION_CODES.O) {
-            if (existingJobId >= 0) {//cancel already running job
-                jobScheduler.cancel(existingJobId);
-                StorageHelper.putInt(context, Constants.PF_JOB_ID, -1);
-            }
-
-            config.getLogger()
-                    .debug(config.getAccountId(), "Push Amplification feature is not supported below Oreo");
+            config.getLogger().debug(config.getAccountId(), "Pushamp feature is not supported below Oreo");
             return;
         }
 
-        if (jobScheduler == null) {
-            return;
-        }
+        String existingWorkName = StorageHelper.getString(context, Constants.PF_WORK_ID, "");
         int pingFrequency = getPingFrequency(context);
 
-        if (existingJobId < 0 && pingFrequency < 0) {
-            return; //no running job and nothing to create
-        }
-
-        if (pingFrequency < 0) { //running job but hard cancel
-            jobScheduler.cancel(existingJobId);
-            StorageHelper.putInt(context, Constants.PF_JOB_ID, -1);
+        // no running work and nothing to create
+        if (existingWorkName.equals("") && pingFrequency <= 0) {
+            config.getLogger()
+                    .debug(config.getAccountId(), "Pushamp - There is no running work and nothing to create");
             return;
         }
 
-        ComponentName componentName = new ComponentName(context, CTBackgroundJobService.class);
-        boolean needsCreate = (existingJobId < 0 && pingFrequency > 0);
-
-        //running job, no hard cancel so check for diff in ping frequency and recreate if needed
-        JobInfo existingJobInfo = getJobInfo(existingJobId, jobScheduler);
-        if (existingJobInfo != null
-                && existingJobInfo.getIntervalMillis() != pingFrequency * Constants.ONE_MIN_IN_MILLIS) {
-            jobScheduler.cancel(existingJobId);
-            StorageHelper.putInt(context, Constants.PF_JOB_ID, -1);
-            needsCreate = true;
+        // Running work exists but hard cancel
+        if (pingFrequency <= 0) {
+            config.getLogger()
+                    .debug(config.getAccountId(), "Pushamp - Cancelling worker as pingFrequency <=0 ");
+            stopWorker();
+            return;
         }
 
-        if (needsCreate) {
-            int jobid = config.getAccountId().hashCode();
-            JobInfo.Builder builder = new JobInfo.Builder(jobid, componentName);
-            builder.setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY);
-            builder.setRequiresCharging(false);
-
-            builder.setPeriodic(pingFrequency * Constants.ONE_MIN_IN_MILLIS, 5 * Constants.ONE_MIN_IN_MILLIS);
```

Comment:
@darshanclevertap @piyush-kukadiya We have 5 minutes specified as the flex interval for the job. Any specific reason and do we need the same for work manager?

Reply:
@Anush-Shand `pingFrequency` comes from the server under the key `pf` which is set as 240. So in production, the actual value is 240 mins or 4 hours. 5 mins is the fallback option in case the server does not send it.

Reply:
@darshanclevertap pingfrequency is defaulted to 4 hours in the sdk as well. The second parameter (currently 5 minutes) specifies the flex interval i.e the window in the period (4 hours) in which the Job/Work can be completed
https://developer.android.com/develop/background-work/background-tasks/persistent/getting-started/define-work#flexible_run_intervals

Reply:
@Anush-Shand Ah, yes. So `setPeriodic` has interval and flex. If you do not provide flex, it will run anytime within the given interval time. We wanted it to be close to every 4 hours, so we set the flex as 5 mins. Hence the Job will run every 4 hours in the last 5 mins. This will make it truly periodic

Reply:
@darshanclevertap 
Got it, we can add the same for work manager. But it reduces the chances of the work actually being completed for that period since there is only a window of 5 minutes to match the constraints

Reply:
@Anush-Shand I think we can do with that. Since it will work in the next run even if one interval is missed.